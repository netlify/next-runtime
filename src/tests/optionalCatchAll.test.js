// Test that next-on-netlify does not crash when pre-rendering index.js file.
// See: https://github.com/netlify/next-on-netlify/issues/2#issuecomment-636415494

const { parse, join } = require('path')
const { readFileSync } = require('fs-extra')
const buildNextApp = require('./helpers/buildNextApp')

// The name of this test file (without extension)
const FILENAME = parse(__filename).name

// The directory which will be used for testing.
// We simulate a NextJS app within that directory, with pages, and a
// package.json file.
const PROJECT_PATH = join(__dirname, 'builds', FILENAME)

// The directory that contains the fixtures, such as NextJS pages,
// NextJS config, and package.json
const FIXTURE_PATH = join(__dirname, 'fixtures')

// Capture the output to verify successful build
let buildOutput

beforeAll(
  async () => {
    buildOutput = await buildNextApp()
      .forTest(__filename)
      .withPages('pages-with-optionalCatchAll')
      .withNextConfig('next.config.js')
      .withPackageJson('package.json')
      .build()
  },
  // time out after 180 seconds
  180 * 1000,
)

describe('next-on-netlify', () => {
  test('builds successfully', () => {
    expect(buildOutput).toMatch('Next on Netlify')
    expect(buildOutput).toMatch('Success! All done!')
  })
})

describe('Routing', () => {
  test('creates Netlify redirects', async () => {
    // Read _redirects file
    const contents = readFileSync(join(PROJECT_PATH, 'out_publish', '_redirects'))
    let redirects = contents.toString()

    // Replace non-persistent build ID with placeholder
    redirects = redirects.replace(/\/_next\/data\/[^\/]+\//g, '/_next/data/%BUILD_ID%/')

    // Check that redirects match
    expect(redirects).toMatchSnapshot()
  })
})
