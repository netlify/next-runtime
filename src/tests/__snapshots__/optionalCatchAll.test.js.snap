// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routing creates Netlify redirects 1`] = `
"# Next-on-Netlify Redirects
/_next/data/%BUILD_ID%/page.json  /.netlify/functions/next_page  200
/page  /.netlify/functions/next_page  200
/_next/data/%BUILD_ID%/index.json  /.netlify/functions/next_all  200
/_next/data/%BUILD_ID%/*  /.netlify/functions/next_all  200
/_next/image*  url=:url w=:width q=:quality  /.netlify/functions/next_image?url=:url&w=:width&q=:quality  200
/  /.netlify/functions/next_all  200
/_next/*  /_next/:splat  200
/*  /.netlify/functions/next_all  200"
`;
